openslides/users/views.py

```diff
@@ -478,3 +162,58 @@ def get_form_kwargs(self):
         kwargs = super().get_form_kwargs()
         kwargs['user'] = self.request.user
         return kwargs
+
+
+class UserLoginView(AjaxMixin, View):
+    """
+    Login the user via ajax.
+    """
+
+    def get_ajax_context(self, **context):
+        if self.success:
+            context.update({
+                'success': True,
+                'user_id': self.user.pk})
+        else:
+            context.update({'success': False})
+
+        return super().get_ajax_context(**context)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            data = json.loads(request.body.decode('utf-8'))
+        except ValueError:
+            self.success = False
+        else:
+            form = AuthenticationForm(request, data=data)
+            if form.is_valid():
+                self.user = form.get_user()
+                auth_login(request, self.user)
+                self.success = True
+            else:
+                self.success = False
+        return self.ajax_response()
+
+
+class UserLogoutView(AjaxMixin, View):
+    """
+    Logout the user via ajax.
+    """
+
+    def post(self, request, *args, **kwargs):
+        auth_logout(request)
+        return self.ajax_response()
+
+
+class WhoAmIView(AjaxMixin, View):
```

Comment:
May be use a View from Django REST Framework.There you can choose between several HTTP requests and decide, what to do so we don't have to use our own AjaxView or Mixin.